### ISSUE TITLE
More detailed outline/document symbol information

### ISSUE THREAD

**Issue by rapgenic** [clangd/clangd#601]
The LSP supports an additional parameter to the `Symbol[]` response to the `documentSymbol` request, which is used to retrieve outline and all the document symbols. This currently unused parameter (`detail`) might be useful to convey some more information about the various symbols for the file. For example:

* The signature for functions
* The type for variables
* The base classes for classes and structs
* maybe more that i did not think of...

To demonstrate this I have made a prototype clangd build in which I have been able to implement the first two points that i described before (I could not reliably get the base classes without crashing the server sometimes) and this is the resulting outline for comparison:

![comparison](https://user-images.githubusercontent.com/10487071/100209238-ebab1100-2f09-11eb-8183-a207f03b9608.png)

I think this might be a useful addition because:
* Reading a signature directly in the outline might be useful for differentiating overloaded functions
* Knowing the type of a variable in addition to its name helps to identify it quicker
* Those details are added in a non too intrusive manner
* Those details are also visible in the breadcrumbs

Right now I am working on the prototype and I am having a few difficulties grasping the internal structure of clang and its classes, which means that I am not sure my code is valid in any case.. So I wanted to report it as a feature request at least to keep track of it and have some discussion. I can provide the code if anyone is interested of course.

**Comment by HighCommander4** [comment]
I agree this would be useful. Note, we already have #520 on file for the function signature part and there's some discussion of it there.

If you'd like to work towards contributing a patch for this, and run into questions, I'm happy to try and answer. It's probably easiest to ask them in the #clangd channel of the [LLVM Discord](https://discord.com/invite/xS7Z362), but you're welcome to ask questions here in the issue as well.

**Comment by HighCommander4** [comment]
There were improvements made in this area in https://reviews.llvm.org/D96751. @rapgenic do you feel they have addressed this issue sufficiently?
